app/graphql/mutations/profile/edit.rb

```diff
@@ -7,22 +7,33 @@ class Edit < BaseMutation
       graphql_name 'UpdateProfile'
 
       argument :id, ID, required: true
+      argument :name, String, required: false
+      argument :description, String, required: false
       argument :compliance_threshold, Float, required: false
       argument :business_objective_id, ID, required: false
+
       field :profile, Types::Profile, null: true
 
       def resolve(args = {})
-        profile = Pundit.authorize(
-          context[:current_user],
-          ::Profile.find(args[:id]),
-          :edit?
-        )
+        profile = authorized_profile(args)
         profile.update(
+          name: args[:name] || profile.name,
+          description: args[:description] || profile.description,
           compliance_threshold: args[:compliance_threshold],
           business_objective_id: args[:business_objective_id]
         )
```

Comment:
I think this should work to only update arguments that are passed in graphql:

```suggestion
        profile.update(args.compact)
```

Reply:
YES! Works like a charm and so much nicer! Thank you!